### PR TITLE
Wam proxy implementation

### PR DESCRIPTION
**Pull request by abhijeetk**

### CODE REVIEW

File: src/agl/WebAppManagerServiceAGL.h [hunk 1]
```diff
@@ -19,7 +19,10 @@ class WebAppManagerServiceAGL : public WebAppManagerService {
     bool initializeAsHostClient();
 
     bool isHostServiceRunning();
-    void setStartupApplication(const std::string& startup_app_id, const std::string& startup_app_uri, int startup_app_surface_id);
+
+    void setStartupApplication(const std::string& startup_app_id,
+        const std::string& startup_app_uri, int startup_app_surface_id,
+        const std::string& startup_proxy_rules = "");
```
Comment (jdapena) [inline src/agl/WebAppManagerServiceAGL.h:25]: const std::string& startup_proxy_rules = std::string() could be also fine.

File: src/agl/WebRuntimeAGL.h [hunk 2]
```diff
@@ -40,10 +41,22 @@ class Launcher {
...
+private:
+  int port_;
+};
+
 class SharedBrowserProcessWebAppLauncher : public Launcher {
 public:
   int launch(const std::string& id, const std::string& uri) override;
   int loop(int argc, const char** argv, volatile sig_atomic_t& e_flag) override;
+private:
+  std::unique_ptr<TinyProxy> tiny_proxy;
```
Comment (jdapena) [inline src/agl/WebRuntimeAGL.h:59]: We could consider using the Qt coding standard as widely adopted in WAM, or move already to the one we expect in the future (Chromium coding style).

I'd be happier with newer chromium style as we expect to also do QtLess, so it'd be tiny_proxy_

File: src/platform/webengine/BlinkWebViewProfileHelper.cpp [hunk 3]
```diff
@@ -17,7 +17,8 @@
 
 #include "BlinkWebViewProfileHelper.h"
 #include "webos/webview_profile.h"
-#include <string.h>
+
+#include <cstring>
```
Comment (jdapena) [inline src/platform/webengine/BlinkWebViewProfileHelper.cpp:21]: Why cstring here?

**Review by jdapena — DISMISSED** [review]
Could you also squash already with the original patch? We'll want to land everything in one go.

File: src/platform/webengine/BlinkWebProcessManager.cpp [hunk 4]
```diff
@@ -120,6 +120,6 @@ int BlinkWebProcessManager::maskForBrowsingDataType(const char* type)
     return BlinkWebViewProfileHelper::maskForBrowsingDataType(type);
 }
 
-void BlinkWebProcessManager::setProxyRules(const char* proxyRules) {
-    return BlinkWebViewProfileHelper::setProxyRules(proxyRules);
+void BlinkWebProcessManager::setProxyRules(const std::string& proxy_rules) {
+    return BlinkWebViewProfileHelper::setProxyRules(proxy_rules);
```
Comment (zhani) [inline src/platform/webengine/BlinkWebProcessManager.cpp:124]: setProxyRules is void function. No return needed

File: src/platform/webengine/BlinkWebProcessManager.cpp [hunk 5]
```diff
@@ -119,3 +119,7 @@ int BlinkWebProcessManager::maskForBrowsingDataType(const char* type)
 {
     return BlinkWebViewProfileHelper::maskForBrowsingDataType(type);
 }
+
+void BlinkWebProcessManager::setProxyRules(const char* proxyRules) {
+    return BlinkWebViewProfileHelper::setProxyRules(proxyRules);
```
Comment (zhani) [inline src/platform/webengine/BlinkWebProcessManager.cpp]: no return needed

File: src/platform/webengine/BlinkWebViewProfileHelper.cpp (hunk 3 above)
Comment (abhijeetk) [reply src/platform/webengine/BlinkWebViewProfileHelper.cpp:21]: It is used for string comparison using  strcmp

**Pull request closed without merging**